File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/question/CreateQuestionRequest.java
package org.example.olympic_ot_project.dto.exam.question;

import jakarta.validation.constraints.*;
import lombok.Data;
import org.example.olympic_ot_project.core.QuestionLevel;
import org.example.olympic_ot_project.core.QuestionType;
import org.example.olympic_ot_project.dto.exam.option.CreateQuestionOptionRequest;

import java.util.List;

@Data
public class CreateQuestionRequest {
    @NotBlank(message = "Nội dung không được để trống")
    private String content;

    @NotNull(message = "Loại câu hỏi không được để trống")
    private QuestionType type;

    @NotNull(message = "Cấp độ câu hỏi không được để trống")
    private QuestionLevel level;

    private String answer;

    @NotNull(message = "Điểm số câu hỏi không được để trống")
    private Integer score;

    @NotNull(message = "Loại câu hỏi không được để trống")
    private Integer categoryId;

    @NotNull(message = "Thời gian trả lời không được bỏ trống")
    @Min(value = 30, message = "Thời gian thấp nhất là 30 giây")
    @Max(value = 120, message = "Thời gian cao nhất là 120 giây")
    private Integer timeLimit;

    private String imageUrl;
    private String videoUrl;

    private List<CreateQuestionOptionRequest> options;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/question/create_question_request.dart
import 'package:olympic_it_project/core/question_level.dart';
import 'package:olympic_it_project/core/question_type.dart';
import 'package:olympic_it_project/dto/admin_manager/question/create_question_option_request.dart';

class CreateQuestionRequest {
  final String content;
  final QuestionType type;
  final QuestionLevel level;
  final String? answer;
  final int score;
  final int categoryId;
  final int? timeLimit;
  final String? imageUrl;
  final String? videoUrl;
  final List<CreateQuestionOptionRequest> options; 

  CreateQuestionRequest({
    required this.content,
    required this.type,
    required this.level,
    this.answer,
    required this.score,
    required this.timeLimit,
    required this.categoryId,
    this.imageUrl,
    this.videoUrl,
    required this.options,
  });

  Map<String, dynamic> toJson() {
    return {
      "content": content,
      "type": type.name,
      "level": level.name,
      "answer": answer,
      "score": score,
      "timeLimit" : timeLimit,
      "categoryId": categoryId,
      "imageUrl": imageUrl,
      "videoUrl": videoUrl,
      "options": options.map((option) => option.toJson()).toList(),
    };
  }
}